src/Workspaces/Core/Portable/Workspace/Solution/SolutionState_Checksum.cs

```diff
@@ -67,50 +74,51 @@ public async Task<Checksum> GetChecksumAsync(CancellationToken cancellationToken
     }
 
     /// <summary>Gets the checksum for only the requested project (and any project it depends on)</summary>
-    public async Task<SolutionStateChecksums> GetStateChecksumsAsync(
-        ProjectId projectId,
+    public async Task<(SolutionStateChecksums checksums, ProjectCone projectCone)> GetStateChecksumsAsync(
+        ProjectId projectConeId,
         CancellationToken cancellationToken)
     {
-        Contract.ThrowIfNull(projectId);
+        Contract.ThrowIfNull(projectConeId);
 
-        AsyncLazy<SolutionStateChecksums>? checksums;
+        AsyncLazy<(SolutionStateChecksums checksums, ProjectCone projectCone)>? checksums;
         lock (_lazyProjectChecksums)
         {
-            if (!_lazyProjectChecksums.TryGetValue(projectId, out checksums))
+            if (!_lazyProjectChecksums.TryGetValue(projectConeId, out checksums))
             {
-                checksums = Compute(projectId);
-                _lazyProjectChecksums.Add(projectId, checksums);
+                checksums = AsyncLazy.Create(static async (arg, cancellationToken) =>
+                {
+                    var (checksums, projectCone) = await arg.self.ComputeChecksumsAsync(arg.projectConeId, cancellationToken).ConfigureAwait(false);
+                    Contract.ThrowIfNull(projectCone);
+                    return (checksums, projectCone.Value);
+                }, arg: (self: this, projectConeId));
+                _lazyProjectChecksums.Add(projectConeId, checksums);
             }
         }
 
         var collection = await checksums.GetValueAsync(cancellationToken).ConfigureAwait(false);
         return collection;
-
-        // Extracted as a local function to prevent delegate allocations when not needed.
-        AsyncLazy<SolutionStateChecksums> Compute(ProjectId projectConeId)
-        {
-            return AsyncLazy.Create(static (arg, c) =>
-                arg.self.ComputeChecksumsAsync(arg.projectConeId, c),
-                arg: (self: this, projectConeId));
-        }
```

Comment:
inlined this.  no need for the local function since Todd moved this to static lambdas.